### PR TITLE
LIT integration tests: boilerplate and the first test

### PR DESCRIPTION
**Pull request by stanislaw**

### CODE REVIEW

File: infrastructure/helpers/lit-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+- name: Run Mull examples
```
Comment (stanislaw) [inline infrastructure/helpers/lit-tests.yaml]: How can I run just this helper file to iterate on it locally?

File: tests-lit/tools/FileCheck.py [hunk 2]
```diff
@@ -0,0 +1,385 @@
+#!/usr/bin/env python3
```
Comment (stanislaw) [inline tests-lit/tools/FileCheck.py]: @AlexDenisov Are you comfortable with having FileCheck.py as source file, not as Python package for now?

File: tests-lit/tools/compilation_database.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#!/usr/local/bin/python3
```
Comment (stanislaw) [inline tests-lit/tools/compilation_database.py]: This is exactly what the #629 is about. I would be fine with having this for now. And then discarding it when Mull starts to support that compilation database format.

File: tests-lit/tools/FileCheck.py (hunk 2 above)
Comment (stanislaw) [reply tests-lit/tools/FileCheck.py]: Another option is to keep the binaries for now. Just let me know what you think.

Comment (AlexDenisov) [reply tests-lit/tools/FileCheck.py]: Python source is definitely better than a binary.

File: infrastructure/helpers/lit-tests.yaml (hunk 1 above)
Comment (AlexDenisov) [reply infrastructure/helpers/lit-tests.yaml]: Exactly as described in the scenario:
```
cd $SOURCE_DIR/tests-lit
export CLANG_EXEC=$LLVM_DIR/bin/clang++
make run_cxx
```

**Pull request merged**